### PR TITLE
Add VGA text interfaces

### PR DESCRIPTION
**Pull request by Sploder12**
Adds a way to print text to arbitrary places on the screen.
Adds ability to clear the screen.

### CODE REVIEW

File: src/os/video/VGA_text.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+// returns a VGA_Char with the supplied attributes
+VGA_Char getVGAchar(unsigned char chr, VGA_Color foreground, VGA_Color background);
+
+
+// prints text at location, will NOT wrap
+void writeText(const char* str, int x, int y, VGA_Color color);
+
+
+// effect cursor
```
Comment (bicknrown) [inline src/os/video/VGA_text.h:41]: affect? (impact)

File: src/os/main.c [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#include "video/VGA_text.h"
+
```
Comment (bicknrown) [inline src/os/main.c:2]: i support the reorganization to separating functionality, in that same way, i think we could have a scheme like:
```
src/boot
src/os
src/lib/video
```
that may make it a bit cleaner to separate how we define what is what in terms of features.

Comment (bicknrown) [reply src/os/main.c:2]: we can also split our documentation that way, to make it nicer to navigate

File: src/os/video/VGA_text.h [hunk 3]
```diff
@@ -0,0 +1,57 @@
+#ifndef VGA_TEXT_H
+#define VGA_TEXT_H
+
```
Comment (bicknrown) [inline src/os/video/VGA_text.h:3]: I appreciate you commenting the header as well as the implementation, this should be a standard we create (or borrow from a system that can create documentation automagically more likely) and then we could create documentation based upon the forward declarations.

File: src/os/video/VGA_text.h (hunk 1 above)
Comment (Sploder12) [reply src/os/video/VGA_text.h]: don't tell anyone I don't know the difference!

**Review by bicknrown — APPROVED** [review]
thanks :   )

**Pull request merged**